http/rest-client-reactive/src/main/java/io/quarkus/ts/http/restclient/reactive/files/FileClientResource.java

```diff
@@ -2,36 +2,55 @@
 
 import java.io.IOException;
 import java.nio.file.Files;
+import java.util.LinkedList;
+import java.util.List;
+import java.util.Optional;
 
+import javax.enterprise.event.Observes;
 import javax.inject.Inject;
 import javax.ws.rs.GET;
 import javax.ws.rs.POST;
 import javax.ws.rs.Path;
 
+import org.eclipse.microprofile.config.inject.ConfigProperty;
 import org.eclipse.microprofile.rest.client.inject.RestClient;
 
+import io.quarkus.runtime.ShutdownEvent;
 import io.smallrye.common.annotation.Blocking;
 import io.smallrye.mutiny.Uni;
 
 @Path("/file-client")
 public class FileClientResource {
     private static final String BIGGER_THAN_TWO_GIGABYTES = OsUtils.SIZE_2049MiB;
-    private final java.nio.file.Path FILE = Files.createTempFile("upload", ".txt").toAbsolutePath();
+
+    private final java.nio.file.Path file;
+    private final List<java.nio.file.Path> deathRow = new LinkedList<>();
     private final FileClient client;
     private final OsUtils utils;
 
     @Inject
-    public FileClientResource(@RestClient FileClient client) throws IOException {
+    public FileClientResource(@RestClient FileClient client,
+            @ConfigProperty(name = "client.filepath") Optional<String> folder) {
```

Comment:
```suggestion
            @ConfigProperty(name = "client.filepath") Optional<String> folder) {
```
Rather than add an Optional as a parameter in a method, I would suggest using a Raw object as `String folder` and then in the method wrap this object with an optional

example:

```
```suggestion
            @ConfigProperty(name = "client.filepath") String folder) {
            utils = OsUtils.get();
        file = Optional.ofNullable(folder)
                .map(existing -> java.nio.file.Path.of(existing).resolve("upload.txt").toAbsolutePath())
                .orElseGet(() -> {
```
Well to me the important part is remove the optional from the method params. 
More details: https://www.javacodegeeks.com/2018/11/java-optional-parameters.html

Reply:
Unfortunately, raw String require property to be provided[1]. Since we only provide this property in LargeFileHandlingIT, and not in the others, this will not work in our case.

[1] https://quarkus.io/guides/config-reference#inject